app/controllers/calculator_constants_controller.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class CalculatorConstantsController < ApplicationController
+  def index
+    @calculator_constants = CalculatorConstant.all
+  end
+
+  def update
+    upload = Upload.where(csv_type: 'CalculatorConstant').first
+    params.each do |key, value|
+      next unless CalculatorConstant.exists?(name: key)
+
+      constant_field = CalculatorConstant.where(name: key).first
+      constant_field.update(float_value: value)
+      update_upload_timestamp(upload, constant_field.updated_at)
+    end
+    flash.notice = 'Calculator Constants have been updated.'
+    redirect_to action: :index
+  end
+end
+
+private
+
+def update_upload_timestamp(upload, updated_timestamp)
+  upload.update(updated_at: updated_timestamp, created_at: updated_timestamp) if updated_timestamp > upload.updated_at
```

Comment:
This shouldn't update the `created_at` column, additionally `completed_at` and `user_id` should be updated.

Can you update the `comment` column as well to indicate what fields were updated or a generic message to indicate that CalculatorConstants were updated through this process?

Reply:
The `created_at` is what is being used in  the upload table on the dashboard. One of the requirements for this story was to update that displayed date when changes are made. 

I'll go ahead and add the changes to update the other two fields as well as create a generic message for the comment column.

Reply:
upload is no longer being updated by calculator constants management table.